storagemarket/provider.go

```diff
@@ -89,9 +89,9 @@ func NewProvider(repoRoot string, dealsDB *db.DealsDB, fullnodeApi v1api.FullNod
 		fullnodeApi: fullnodeApi,
 		db:          dealsDB,
 
-		acceptDealsChan:  make(chan acceptDealReq, 128),
-		failedDealsChan:  make(chan failedDealReq, 128),
-		restartDealsChan: make(chan restartReq, 128),
+		acceptDealsChan:  make(chan acceptDealReq),
+		failedDealsChan:  make(chan failedDealReq),
+		restartDealsChan: make(chan restartReq),
```

Comment:
Changing these channels from buffered to unbuffered - the rational is - if we have a bug with respect to the synchronicity of how we consume values from these channels, we'll only notice a bug after contention happens on one of them, which might be at high load - so better make the channel unbuffered, and if it is necessary for them to be buffered, we should reduce 128 to a much lower value.